pgjdbc/src/main/java/org/postgresql/util/HostSpec.java

```diff
@@ -29,6 +32,10 @@ public String toString() {
     return host + ":" + port;
   }
 
+  public URI toURI() throws URISyntaxException {
+    return new URI("socket", null, host, port, null, null, null);
```

Comment:
This is the format expected by the [DefaultProxySelector](http://hg.openjdk.java.net/jdk8u/jdk8u20/jdk/file/f5d77a430a29/src/share/classes/sun/net/spi/DefaultProxySelector.java#l190)